### PR TITLE
Changes way of handling timeouts

### PR DESCRIPTION
**Pull request by SilverSheep**

### CODE REVIEW

**Review by bfugas — CHANGES_REQUESTED** [review]

File: src/main/java/ai/applica/spring/boot/starter/temporal/annotations/ActivityStub.java [hunk 1]
```diff
@@ -23,16 +23,16 @@
...
+ * <p>One must specify duration string according to java.time.Duration format.
  */
 @Retention(RetentionPolicy.RUNTIME)
 @Target({ElementType.FIELD})
 public @interface ActivityStub {
 
-  long duration() default -1;
-
-  String durationUnits() default "SECONDS";
+  String duration() default "PT0S";
```
[the remark this answers is not in the document]
Comment (SilverSheep) [reply src/main/java/ai/applica/spring/boot/starter/temporal/annotations/ActivityStub.java:35]: As discussed - for now I'll leave it as it is

File: src/main/java/ai/applica/spring/boot/starter/temporal/processors/ActivityStubInterceptor.java [hunk 2]
```diff
@@ -49,6 +48,7 @@
 @Slf4j
 @RequiredArgsConstructor
 public class ActivityStubInterceptor {
+  private static final String defaultDuration = "PT0S";
```
[the remark this answers is not in the document]
Comment (SilverSheep) [reply src/main/java/ai/applica/spring/boot/starter/temporal/processors/ActivityStubInterceptor.java]: done

File: src/main/java/ai/applica/spring/boot/starter/temporal/processors/ActivityStubInterceptor.java [hunk 3]
```diff
@@ -97,19 +97,14 @@ private ActivityOptions buildOptions(
...
     // from default method
     options = temporalOptionsConfiguration.modifyDefaultActivityOptions(options);
-    if (activityStubAnnotation.duration() != -1) {
-      options.setScheduleToCloseTimeout(
-          Duration.of(
-              activityStubAnnotation.duration(),
-              ChronoUnit.valueOf(activityStubAnnotation.durationUnits())));
+    String duration = activityStubAnnotation.duration();
+    if (!defaultDuration.equals(duration)) {
+      options.setStartToCloseTimeout(Duration.parse(duration));
```
[the remark this answers is not in the document]
Comment (SilverSheep) [reply src/main/java/ai/applica/spring/boot/starter/temporal/processors/ActivityStubInterceptor.java:107]: As discussed - leaving it as it is

File: src/test/java/ai/applica/spring/boot/starter/temporal/samples/HelloPropertiesTest.java [hunk 4]
```diff
@@ -84,10 +80,10 @@ public void testTimeoutsFromPropertiesAreReadCorrectly() {
     dotWorker.registerActivitiesImplementations(timeoutActivity);
     testEnv.start();
 
-    Duration timeout = workflow.getTimeout();
-    assertThat(timeout).isEqualTo(Duration.ofSeconds(1));
+    //    Duration scheduleToCloseTimeout = workflow.getScheduleToCloseTimeout();
```
[the remark this answers is not in the document]
Comment (SilverSheep) [reply src/test/java/ai/applica/spring/boot/starter/temporal/samples/HelloPropertiesTest.java]: Removed comments

File: src/test/java/ai/applica/spring/boot/starter/temporal/samples/apps/HelloProperties.java [hunk 5]
```diff
@@ -83,18 +96,31 @@ public Duration getTimeout() {
...
+    //    public Duration getScheduleToCloseTimeout() {
+    //      return Activity.getExecutionContext().getInfo().getScheduleToCloseTimeout();
+    //    }
+
     @Override
-    public Duration getTimeout() {
-      return Activity.getExecutionContext().getInfo().getScheduleToCloseTimeout();
+    public Duration getStartToCloseTimeout() {
+      // TODO: Find equivalent to...
+      //      Activity.getExecutionContext().getInfo().getScheduleToStartTimeout();
```
[the remark this answers is not in the document]
Comment (SilverSheep) [reply src/test/java/ai/applica/spring/boot/starter/temporal/samples/apps/HelloProperties.java]: I couldn't find a way to test it

File: src/test/resources/application.yml [hunk 6]
```diff
@@ -22,14 +22,12 @@ spring.temporal:
...
-      scheduleToCloseTimeout: 20
-      scheduleToCloseTimeoutUnit: SECONDS
+      scheduleToCloseTimeout: PT20S
     PropertiesActivity:
-      scheduleToCloseTimeout: 1
-      scheduleToCloseTimeoutUnit: SECONDS
+      scheduleToCloseTimeout: PT1S
     "[PropertiesDotWorkflow.PropertiesActivity]":
-      scheduleToCloseTimeout: 2
-      scheduleToCloseTimeoutUnit: SECONDS
```
[the remark this answers is not in the document]
Comment (SilverSheep) [reply src/test/resources/application.yml:32]: Done

File: src/main/java/ai/applica/spring/boot/starter/temporal/annotations/ActivityStub.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (bfugas) [reply src/main/java/ai/applica/spring/boot/starter/temporal/annotations/ActivityStub.java:35]: :+1:

File: src/main/java/ai/applica/spring/boot/starter/temporal/processors/ActivityStubInterceptor.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (bfugas) [reply src/main/java/ai/applica/spring/boot/starter/temporal/processors/ActivityStubInterceptor.java:107]: `!defaultDuration.equals(duration)` solves it :+1:

**Review by bfugas — APPROVED** [review]

**Review by bfugas — APPROVED** [review]

**Pull request merged**